### DISCUSSION TITLE
How Modern MIDI Mixing could work in CorsixTH (Fluidsynth, Using Internal MIDI Selection)

### DISCUSSION THREAD

**Discussion by ravenPenfold** [CorsixTH/CorsixTH#2269] [Research Room]
In DOSBOX, you are able to list all avalible midi devices with `mixer /listmidi`, and it is entirely possible to implement internal settings midi-selection tools and lists without having to rely on external options. 
![image](https://user-images.githubusercontent.com/110283510/195574929-94b37fcc-9cba-4a44-ab61-2d528f097449.png)

It is also possible to use Fluidsynth and switch between the two (via a toggle), thus allowing for the options between a MIDI Device such as a Sound-Blaster 16, the bog-standard GM.DLS, or even the SOUND-Canvas 88.

In terms of configuring this, if we have 2 flags - one for defining whether to use Fluidsynth (Soundfonts), or to use a MIDI Device or Software (e.g SYXG50), and another to configure which device or soundfont should be used based on the former configuration.

![image](https://user-images.githubusercontent.com/110283510/195575289-cf67c3c1-d991-4635-86ff-babc2f410471.png)

This approach to allowing for easy user MIDI Choice without having to resort to installing software such as VirtualMidiSynth would be more viable in the future, and make choosing the instruments you hear easier.

(also, there may or may not be a default non-GM.dls soundfont in the works for this if we do use Fluidsynth)

**Comment by ravenPenfold** [comment]
Investigation into possible SF2 files we could use, if not our own:

**Timbres of Heaven** - https://midkar.com/soundfonts/

will be edited/added on as I find more

**Comment by TheCycoONE** [comment]
Nice. We are now using fluidsynth on windows (though still through SDL2_Mixer) and I've familiarized myself with the APIs. I believe we could pivot to bypassing SDL2_Mixer in a future release. In Windows you will be able to specify your own soundfonts but not yet native midi devices/ports.
